### PR TITLE
Add avatar component

### PR DESCRIPTION
**Pull request by tysongach**
[View prototype on CodePen][codepen].

[codepen]: https://codepen.io/tysongach/pen/XQqWPK/

Screen Shot 2019-04-18 at 21 07 49

Closes https://github.com/thoughtbot/design-system/issues/63

### CODE REVIEW

File: src/avatar/lib/avatar.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.tbds-avatar {
+  --border-radius: 3px;
+  --size: 3rem;
+
+  border-radius: var(--border-radius);
+  display: inline-block;
+  height: var(--size);
+  object-fit: cover;
```
Comment (setphen) [inline src/avatar/lib/avatar.scss]: regarding CSS variables and object-fit: Should we describe somewhere in the package what browsers support tbds?

File: src/avatar/lib/avatar.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.tbds-avatar {
+  --border-radius: 3px;
+  --size: 3rem;
```
Comment (setphen) [inline src/avatar/lib/avatar.scss]: Is this the pattern you were suggesting in https://github.com/thoughtbot/design-system/pull/71#discussion_r275511258 ?

File: src/avatar/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Avatar
+
+[View prototype on CodePen][codepen].
```
Comment (setphen) [inline src/avatar/README.md]: Should all components have this sort of prototype?

File: src/avatar/lib/avatar.scss (hunk 1 above)
Comment (tysongach) [reply src/avatar/lib/avatar.scss]: We should, and do!

> tbds supports:
>
>    - Latest versions of Chrome, Firefox, and Safari
>    - Edge 15+


https://github.com/thoughtbot/design-system#browser-support

File: src/avatar/README.md (hunk 3 above)
Comment (tysongach) [reply src/avatar/README.md]: This is the first time I’ve done it, which mostly was driven by a way to actually show the components rendered. Without fleshing out a full documentation site, CodePen links felt like a cheap first stab.

What are you thoughts on it?

File: src/avatar/lib/avatar.scss (hunk 2 above)
Comment (tysongach) [reply src/avatar/lib/avatar.scss]: We could use CSS variables in that scenario, though Sass variables would be fine, too. Mostly my point was just offering it as a configuration point, so a consumer could easily say “I need my icons to be `1.5em`, so I will override this variable to do that”

Your question is valid, though, and we should document why and when to use Sass variables, and CSS variables.

File: src/avatar/README.md (hunk 3 above)
Comment (setphen) [reply src/avatar/README.md]: I think it's great – though we may want to use the thoughtbot Codepen account for maintainability.

Comment (tysongach) [reply src/avatar/README.md]: Definitely, good catch! I will move this component over to the thoughtbot account and use that going forward.

Comment (tysongach) [reply src/avatar/README.md]: Update to use thoughtbot CodePen account.

**Pull request merged**